File: doc-examples/mongo-example-records-java/src/main/java/example/Student.java

package example;

import io.micronaut.data.annotation.GeneratedValue;
import io.micronaut.data.annotation.Id;
import io.micronaut.data.annotation.MappedEntity;
import io.micronaut.data.annotation.Version;
import org.bson.types.ObjectId;

// tag::student[]
@MappedEntity
public record Student(
        @Id
        @GeneratedValue ObjectId id,
        @Version Long version,
        String name) {
}
// end::student[]

File: doc-examples/mongo-example-kotlin-ksp/src/main/kotlin/example/Student.kt
package example

import io.micronaut.data.annotation.*
import org.bson.types.ObjectId

// tag::student[]
@MappedEntity
data class Student(
        @field:Id @GeneratedValue
        val id: ObjectId?,
        @field:Version
        val version: Long?,
// end::student[]

        val name: String,
        @Relation(value = Relation.Kind.MANY_TO_MANY, cascade = [Relation.Cascade.PERSIST])
        val courses: List<Course>?,
        @Relation(value = Relation.Kind.ONE_TO_MANY, mappedBy = "student")
        val ratings: List<CourseRating>?
) {
    constructor(name: String, items: List<Course>) : this(null, null, name, items, emptyList())
}
